tests/docker/globus.debian12.Dockerfile

```diff
@@ -0,0 +1,75 @@
+FROM debian:12 
+
+ARG DEBIAN_FRONTEND=noninteractive
+ARG APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=true
+
+#### install basic packages ####
+RUN apt-get update && apt-get install -y curl \
+    g++ \
+    gcc \
+    ftp \
+    telnet \
+    libcurl4-openssl-dev \
+    unzip \
+    python3 \
+    python3-distro \
+    python3-psutil \
+    python3-jsonschema \
+    python3-requests \
+    python3-pip \
+    python3-pyodbc \
+    wget \
+    gnupg2 \
+    lsb-release \
+    && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/*
+
+#### Get and install iRODS repo ####
+RUN wget -qO - https://packages.irods.org/irods-signing-key.asc | apt-key add - && \
+    echo "deb [arch=amd64] https://packages.irods.org/apt/ $(lsb_release -sc) main" | tee /etc/apt/sources.list.d/renci-irods.list && \
+    apt-get update
+
+#### Install icommands - used to set up, validate and tear down tests. ####
+#### Install externals and dev package to build the connector.         ####
+
+# Comment out the irods-icommands and irods-dev if testing against locally
+# built packages.
+RUN apt-get install -y \
+    irods-icommands \
+    irods-dev \
+    irods-externals-cmake3.21.4-0 \
+    irods-externals-clang13.0.1-0 \
+    irods-externals-clang-runtime13.0.1-0 \
+    && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/*
+
+#### Get and install globus repo ####
+RUN wget -q https://downloads.globus.org/globus-connect-server/stable/installers/repo/deb/globus-repo_latest_all.deb && \
+    apt-get install -y ./globus-repo_latest_all.deb && \
```

Comment:
Always run `apt-get update` before any `apt-get install` commands in `RUN` blocks. If any packages need to be fetched from repositories, the package lists need to be in the cache first. This applies even when installing packages from files on the disk, as they may need to pull package requirements from the repos. This applies to all usages of `apt-get install` in `RUN` blocks.

Reply:
I think I have all of these covered now.